### ISSUE TITLE
Calculate every 15min

### ISSUE THREAD

**Issue by APayerl** [custom-components/nordpool#147]
I have on good authority that at least Sweden concidering changing the pricing from every hour to every 15 minutes. Does this integration support such a change or would it require a big rewright? 
Just thought it would be worth mentioning in good time.. 😅

**Comment by stefancn** [comment]
O shit... can you get some info from what that info is coming from...  because this would have a huge impact on all systems on all parties that handle energy.

**Comment by APayerl** [comment]
Here is a source but I hope you know Swedish, or can use Google translate! 😉
https://www.svk.se/utveckling-av-kraftsystemet/systemansvar--elmarknad/ny-nordisk-balanseringsmodell-nbm/15-minuters-tidsupplosning/

SVK is "Svenska kraftnät" and is the organisation in Sweden that is responsible for the big power cables across Sweden.

**Comment by stefancn** [comment]
Yes I'm from Sweden and a newly certified electric. .  I read a lot of our Laws etc. SVK and EU's rules of energy and transmission. Interesting thou, not a zip about this.... nor the rules of debit protocol across borders.... (eg. the cause why energy is so expensive in IE3 and IE4), thanks for you answer.
/Stefan

**Comment by Hellowlol** [comment]
I dont know how much work it would be

**Comment by APayerl** [comment]
> Yes I'm from Sweden and a newly certified electric. .  I read a lot of our Laws etc. SVK and EU's rules of energy and transmission. Interesting thou, not a zip about this.... nor the rules of debit protocol across borders.... (eg. the cause why energy is so expensive in IE3 and IE4), thanks for you answer.
> /Stefan

Well its still early days... As the link said its not even sure it will be implemented during 2023...

**Comment by milkboy** [comment]
The same will be true for the Finland also (https://www.fingrid.fi/sahkomarkkinat/markkinoiden-yhtenaisyys/pohjoismainen-tasehallinta/varttitase/#taustaa). So the backends should update in May 2023, but the pricing will still be the same for the full hour "until later", whatever that means :wink:

**Comment by WallyDW** [comment]
I guess the trading will end at the same time as today, but the price interval will just be smaller.
As long as you can download the entire day-ahead list, then the smaller intervals should be a really small change to make.

**Comment by honkasalo** [comment]
Yes, this 15min interval is due to EU-regulations. Actually, deadline was already in 2020, but in Finland this will be implemented in 22.5.2023. No mention on Milkboy's linked article (by Fingrid, Finnish distribution network operator) about other Nordic countries deadlines, but since Nordic operators are closely tied, it's probably same time in Sweden and Norway at least?

And yes, trading and pricing seems to be still in 60min period in practice, same price for all four 15min time division, but it will be switched to 15min at some stage. This need also reprogramming of consumer utility meters etc.

**Comment by zeminator** [comment]
Same story in Finland.. Not sure what will be exact date. Because they have informed the stuff so unclear. (example 15min net energy usage was informed to start on autumn, but actually it started already 15 april). But if i have understand correctly during the autumn the 15min prising should start. And of course i think that will happen whole Nordic.

**Comment by dala318** [comment]
https://www.nordpoolgroup.com/en/trading/transition-to-15-minute-market-time-unit-mtu/

**Comment by Ferrox85** [comment]
Recieved this also from my energy company by the 11th of june (as mentioned in the post above). Is this going to be implemented or should I look for other options?

I don't have any experience developing for a hass integration unfortinatly...

I like the nordpool sensor of it's energy company independent info.
